src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArraysTests.cs

```diff
@@ -0,0 +1,444 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class AvoidConstArraysTests
+    {
+        [Fact]
+        public async Task IdentifyConstArrays()
+        {
+            // Implicit initialization check
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+
+public class A
+{
+    public void B()
+    {
+        Console.WriteLine({|CA1850:new[]{ 1, 2, 3 }|});
+    }
+}
+", @"
+using System;
+
+public class A
+{
+    private static readonly int[] value = new[]{ 1, 2, 3 };
+
+    public void B()
+    {
+        Console.WriteLine(value);
+    }
+}
+");
+
+            await VerifyVB.VerifyCodeFixAsync(@"
+Imports System
+
+Public Class A
+    Public Sub B()
+        Console.WriteLine({|CA1850:{1, 2, 3}|})
+    End Sub
+End Class
+", @"
+Imports System
+
+Public Class A
+    Private Shared ReadOnly value As Integer() = {1, 2, 3}
+    Public Sub B()
+        Console.WriteLine(value)
+    End Sub
+End Class
+");
+
+            // Explicit initialization check
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+
+public class A
+{
+    public void B()
+    {
+        Console.WriteLine({|CA1850:new int[]{ 1, 2, 3 }|});
```

Comment:
The API does accept it, it just prints out the type of the argument, `System.Int32[]`, rather than the contents of the array. I just wanted an example that would accept an array as an argument.